viYarn/Episode/model/EpisodesViewModelResponsibilities.swift

```diff
@@ -16,33 +17,83 @@ protocol EpisodeLoaderDelagate: class{
 
 protocol EpisodesViewModelResponsibilities: UITableViewDataSource, UITableViewDelegate {
     var delegate: EpisodeLoaderDelagate? {get set}
-    func loadEpisode()
+    func loadEpisode(storyID: Int)
 }
 
 
 class MockEpisodeViewModel: NSObject, EpisodesViewModelResponsibilities {
     weak var delegate: EpisodeLoaderDelagate?
+    private let networkProvider: EpisodeNetworkProvider
+    let host = "http://ec2-54-234-103-230.compute-1.amazonaws.com"
+    var episodes: Episode?
+    
+    init(networkProvider: EpisodeNetworkProvider) {
+        self.networkProvider = networkProvider
+    }
     
-    var episodes = [Episode]()
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return episodes.count
+        guard let count = episodes?.posts.count else { return 0 }
+        return count
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         if let cell = tableView.dequeueReusableCell(withIdentifier: EpisodeTableViewCell.cellID) as? EpisodeTableViewCell {
-            return cell.configured(for: indexPath.row, with: episodes[indexPath.row])
+            guard let postEpisode = episodes?.posts[indexPath.row] else { return UITableViewCell() }
+            return cell.configured(for: indexPath.row, with: postEpisode)
         }
         
         return UITableViewCell()
     }
     
-    func loadEpisode() {
-        self.episodes = [
-            Episode(title: "title", description: "Elephant"),
-            Episode(title: "title2", description: "Elephant")
-        ]
-        delegate?.didLoadEpisodes()
+    func loadEpisode(storyID: Int) {
+        networkProvider.episodes(withHost: host, storyID: storyID, success: { [unowned self] (loadedEpisodes) in
```

Comment:
p1: withHost: - let's inject that into networkProvider itself. @FlareKin  thoughts?